core/src/main/java/org/infinispan/globalstate/impl/EmbeddedLocalConfigurationManager.java

```diff
@@ -0,0 +1,167 @@
+package org.infinispan.globalstate.impl;
+
+import static org.infinispan.factories.KnownComponentNames.CACHE_DEPENDENCY_GRAPH;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.nio.channels.FileLock;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.api.CacheContainerAdmin;
+import org.infinispan.commons.util.concurrent.ConcurrentHashSet;
+import org.infinispan.configuration.ConfigurationManager;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.configuration.global.GlobalStateConfiguration;
+import org.infinispan.configuration.parsing.ConfigurationBuilderHolder;
+import org.infinispan.configuration.parsing.ParserRegistry;
+import org.infinispan.eviction.PassivationManager;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.globalstate.GlobalConfigurationManager;
+import org.infinispan.globalstate.LocalConfigurationManager;
+import org.infinispan.jmx.CacheJmxRegistration;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.persistence.manager.PersistenceManager;
+import org.infinispan.util.DependencyGraph;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * The default implementation of {@link LocalConfigurationManager} for embedded scenarios.
+ *
+ * This component persists cache configurations to the {@link GlobalStateConfiguration#persistentLocation()} in a
+ * <pre>caches.xml</pre> file which is read on startup.
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public class EmbeddedLocalConfigurationManager implements LocalConfigurationManager {
```

Comment:
for some reason, I was expecting 2 implementations. 
one with persistence supported, and another without it...

Reply:
I didn't feel like a simple ephemeral implementation was necessary...